src/uct/ib/ud/base/ud_iface.c

```diff
@@ -342,9 +399,15 @@ ucs_status_t uct_ud_iface_complete_init(uct_ud_iface_t *iface)
 
 void uct_ud_iface_remove_async_handlers(uct_ud_iface_t *iface)
 {
-    uct_base_iface_progress_disable(&iface->super.super.super,
-                                    UCT_PROGRESS_SEND | UCT_PROGRESS_RECV);
-    ucs_async_remove_handler(iface->async.timer_id, 1);
+    int event_fd;
+
+    uct_ud_iface_progress_disable(&iface->super.super.super,
+                                  UCT_PROGRESS_SEND | UCT_PROGRESS_RECV);
+    if ((iface->async.event_cb != NULL) &&
+        (uct_ib_iface_event_fd_get(&iface->super.super.super,
+                                   &event_fd) == UCS_OK)) {
+        ucs_async_remove_handler(event_fd, 1);
```

Comment:
i think it would looke better with nested if:
```
if (event_cb!=NULL) {
   status = uct_ib_iface_event_fd_get(&event_fd);
   if (status == OK) {
    remove_handler
  }
}
```